models/source_list.py

```diff
@@ -141,14 +145,15 @@ def save(self):
         if self.data is None:
             raise ValueError("Cannot save source list without data")
 
-        filename = self.invent_filename()
+        filename = self.image.filepath
+        if filename is None:
+            filename = self.image.invent_filename()
 
-        if self.is_sub:
-            filename += '_detections'
-        else:
-            filename += '_sources'
+        if filename.endswith(('.fits', '.h5', '.hdf5')):
+            filename = os.path.splitext(filename)[0]
 
         # TODO: use FITS and astropy tables instead?
```

Comment:
The advantage of using FITS is that the sources file will be more easily read by hand by somebody who wants to do ad-hoc stuff.  That is, reading the .npy file is easy, but the FITS table is a bit more self-documenting, generally.  (I guess you could have all named columns and such in the .npy file as well; not sure what the sep returns.)